libs/ui-watt/src/lib/components/table/watt-table.component.ts

```diff
@@ -301,8 +307,12 @@ export class WattTableComponent<T>
         .map((key) => this.columns[key].size)
         .map((size) => size ?? 'auto');
 
-      // Add space for extra checkbox column
-      if (this.selectable) sizing.unshift('var(--watt-space-xl)');
+      if (this.selectable) {
+        // Add space for extra checkbox column
+        sizing.unshift('var(--watt-space-xl)');
+
+        this._selectionModel.setSelection(...this.initiallySelectedRows);
```

Comment:
suggestion: this will reset the selection when `columns` or `displayedColumns` changes, which I don't think we want. perhaps move it out of the `if`, so it only happens when `changes.selectable` is truthy.

Reply:
Good point. Will do some more testing with difference scenarios to see what happens.

Reply:
Tried a few cases and pushed a fix. It becomes a bit tricky when `selectable` is toggled. For now when `selectable` is truthy, the initial selection will be set. Toggling `selectable` afterwards will not cause any selection to be reset.

Reply:
That sounds fine IMO, I can't imagine any scenarios where you would want to toggle `selectable` anyway.

Reply:
Another solution that might cover the use case you mentioned, is to make the checkbox column configurable in the `columns` object. Something like this:

```ts
columns = {
  checkbox: {
    accessor: row => row.enabled
  },
  id: {
    accessor: `id`
  }
}
```

Then if the row data from the server has `enabled` set to `true`, the checkbox would be selected initially. If the data is updated at a later time, the new rows should also be checked according to the `enabled` state.

Just an idea, haven’t thought it all the way through 🙂

Reply:
Sounds interesting. Might be something we can explore in the future :)